### PR TITLE
Use a vec array for local_isrc* for coalesced array access

### PR DESCRIPTION
**Pull request by isuruf**
This gives about a 8% speedup

### CODE REVIEW

**Comment by inducer** [comment]
As discussed: Maybe just transposing the local array can have the same effect.

**Comment by inducer** [comment]
Any chance of turning this into a transform rather than changing the code? In my view, it rather decreases the readability of the code.

File: sumpy/p2p.py [hunk 1]
```diff
@@ -656,6 +656,9 @@ def get_optimized_kernel(self, max_nsources_in_one_box,
             knl = lp.tag_inames(knl, {"itgt_box": "g.0", "inner": "l.0"})
             knl = lp.set_temporary_address_space(knl,
                 ["local_isrc", "local_isrc_strength"], lp.AddressSpace.LOCAL)
+            knl = lp.concatenate_memory_layout_of_temporaries(knl,
```
Comment (inducer) [inline sumpy/p2p.py:659]: Could you add a comment explaining the conjectured explanation for why this helps performance, the observed increase, qualified with the hardware on which you observed it?

Comment (isuruf) [reply sumpy/p2p.py:667]: Done in becfc2f

**Comment by inducer** [comment]
LGTM, thanks!

**Pull request merged**